### DISCUSSION TITLE
Slow at first, then all fine

### DISCUSSION THREAD

**Discussion by aifrog** [open-webui/open-webui#2611] [Q&A]
I installed ollama and openwebui on docker on my windows machine (+GPU). When I execute the first question - for example using llama3 - the first response takes many, many minutes. Then - many minutes later - the GPU in the taskmanager seems to pick up the model using 6GB. Nothing happens. If I then execute the same question again it work very nice and speedy. Same withe other questions, After a while the model is automatically removed from the GPU and the same starts over again. Is this the expected behaviour?

BTW, the following is in the ollama log:
"time=2024-05-27T16:33:43.770Z level=INFO source=server.go:545 msg="llama runner started in 148.57 seconds"

**Comment by justinh-rahb** [comment]
Yes this is expected behaviour. The waiting you experience for the first request is Ollama loading the model from your disk storage to memory, which could be slow if you're storing models on a spinning disk instead of an SSD. The 2nd generation is faster if the model is still loaded into memory. Ollama unloads models after 5 minutes of inactivity by default, leading to a slow followup request after this. You can set the ollama keepalive parameter to `-1` for unlimited, or to values such as `30m` for 30 minutes.

**Reply by kevinhq** [reply]
but, why ollama run on terminal never had any of this issue? this is clearly issue with open web ui. this makes me think twice to use it for production. if on local it needs to wait many minutes, what about users with slow connection if it runs on server accessed through internet?

**Reply by rgaricano** [reply]
I suspect that your problem may be due to how the models are loaded on the CPU/GPU, it looks like the GPU is loading and unloading to the CPU when reaching the limits, you can try configuring the embedding model to be moved directly by ollama and, as you have been told, to keep the models for longer... start by disabling everything that can load the system (audio, rerank,...) and with light models to see how it responds.

**Reply by kevinhq** [reply]
same response with light models. but, even with heaviest model, it loads instantly as long as on terminal (ollama cli). whatever, i give up on open-webui. won't use this anymore so my previous comment is not relevant, i think.
